src/util/logger.rs

```diff
@@ -58,28 +58,29 @@ where
     init_log(filtered, level)
 }
 
-pub fn get_level_by_string(lv: &str) -> Level {
+pub fn get_level_by_string(lv: &str) -> Option<Level> {
     match &*lv.to_owned().to_lowercase() {
-        "critical" => Level::Critical,
-        "error" => Level::Error,
+        "critical" => Some(Level::Critical),
+        "error" => Some(Level::Error),
         // We support `warn` due to legacy.
-        "warning" | "warn" => Level::Warning,
-        "debug" => Level::Debug,
-        "trace" => Level::Trace,
-        "info" | _ => Level::Info,
+        "warning" | "warn" => Some(Level::Warning),
+        "debug" => Some(Level::Debug),
+        "trace" => Some(Level::Trace),
+        "info" => Some(Level::Info),
+        _ => None,
```

Comment:
em, I prefer using info instead of None here. Log Level is not a critical configuration.

Reply:
Returning a `None` here allows the caller to determine if an invalid value was provided, and decide how to act. If we replace this with `Level::Info` we are forcing a default. `Level::Info` is should still be the default after this PR.